File: python/DataController.py
from AnggotaDPR import AnggotaDPR
from Table import Table

class DataController:
    __container = []

    def __init__(self):
        pass


    def getContainer(self):
        return self.__container
    
    def setContainer(self, container):
        self.__container = container

    def searchByID(self, id):
        found = False
        index = 0
        while not found and index < len(self.__container):
            if(self.__container[index].getId() == id):
                found = True
            else:
                index += 1
        
        if found:
            return self.__container[index]
        else:
            return None
        
    def createData(self):
        id = ""
        name = ""
        field = ""
        electoral_district = ""
        position = ""
        party = ""

        print("Please give attrubutes for new data:")
        print("ID                 : ", end='')
        id = input()
        print("Name               : ", end='')
        name = input()
        print("Field              : ", end='')
        field = input()
        print("Electoral District : ", end='')
        electoral_district = input()
        print("Position           : ", end='')
        position = input()
        print("Party              : ", end='')
        party = input()

        anggotaDPR = AnggotaDPR(id, name, field, electoral_district, position, party)
        
        self.__container.append(anggotaDPR)

        print("Data has been successfully created!")

    
    def modifyData(self):
        targetID = ""
        print("PLease select data to modify by ID : ", end='')
        targetID = input()
        target = self.searchByID(targetID)

        if(target is None):
            print("Data was not found!")
            return
        
        print("Data found. Please enter new data attributes")

        name = ""
        field = ""
        electoral_district = ""
        position = ""
        party = ""

        print("Please give attrubutes for new data:")
        print("Name               : ", end='')
        name = input()
        print("Field              : ", end='')
        field = input()
        print("Electoral District : ", end='')
        electoral_district = input()
        print("Position           : ", end='')
        position = input()
        print("Party              : ", end='')
        party = input()

        target.setName(name)
        target.setField(field)
        target.setElectoralDistrict(electoral_district)
        target.setPosition(position)
        target.setParty(party)

        print("Data successfully modified!")

    def deleteData(self):
        
        targetID = ""
        print("Please select data to delete by ID : ", end='')
        targetID = input()
        target = self.searchByID(targetID)

        if(target is None):
            print("Data was not found!")
            return
        
        print("Data has been found.")
        print("{}. {}".format(target.getId(), target.getName()))
        print("Confirm to delete? (y/n)")
        del_confirm = input()
        if del_confirm == 'y':
            self.__container.remove(target)
            print("Data has successfully been deleted")
        else:
            print("Cancelled data deletion")
    

    def showAll(self):
        table = Table()
        data = [["" for j in range(6)] for i in range(len(self.__container) + 1) ] # + 1 untuk diawal diisi dengan header tabel

        data[0][0] = "ID"
        data[0][1] = "Name"
        data[0][2] = "Field"
        data[0][3] = "Electoral District"
        data[0][4] = "Position"
        data[0][5] = "Party"

        for i, anggotaDPR in enumerate(self.__container):
            data[i + 1][0] = anggotaDPR.getId()
            data[i + 1][1] = anggotaDPR.getName()
            data[i + 1][2] = anggotaDPR.getField()
            data[i + 1][3] = anggotaDPR.getElectoralDistrict()
            data[i + 1][4] = anggotaDPR.getPosition()
            data[i + 1][5] = anggotaDPR.getParty()

        table.createRows(len(self.__container) + 1, 6, data)
        print("\n")
                

File: java/DataController.java
import java.util.Scanner;
import java.util.ArrayList;


public class DataController {

    private ArrayList<AnggotaDPR> container = new ArrayList<>();
    private Scanner sc = new Scanner(System.in);


    public DataController() {

    }

    public DataController(ArrayList<AnggotaDPR> src) {
        this.container = src;
    }

    public ArrayList<AnggotaDPR> getContainer() {
        return this.container;
    }

    public void setContainer(ArrayList<AnggotaDPR> container) {
        this.container = container;
    }

    public AnggotaDPR searchById(String id) {
        for(AnggotaDPR obj : this.container) {
            if(obj.getId().equals(id)) {
                return obj;
            }
        }
        return null;
    }

    public void createData() {
        String id, name, field, electoralDistrict, position, party;

        System.out.println("\nPlease give attributes for new data:");
        System.out.print("ID                 : ");
        id = sc.next();
        sc.nextLine();
        System.out.print("Name               : ");
        name = sc.nextLine();
        System.out.print("Field              : ");
        field = sc.nextLine();
        System.out.print("Electoral District : ");
        electoralDistrict = sc.nextLine();
        System.out.print("Position           : ");
        position = sc.nextLine();
        System.out.print("Party              : ");
        party = sc.nextLine();

        AnggotaDPR temp = new AnggotaDPR(id, name, field, electoralDistrict, position, party);
        this.container.add(temp);

        System.out.println("Data has been successfully created!");
        

    }

    public void modifyData() {
        String targetID;
        System.out.print("\nPlease select data to modify by ID : ");
        targetID = sc.next();

        AnggotaDPR target = this.searchById(targetID);
        if(target == null) {
            System.err.println("Data was not found!");
            return; 
        }

        System.out.println("\nData found. Please enter new data attributes");
        sc.nextLine();

        String name, field, electoralDistrict, position, party;

        System.out.print("Name               : ");
        name = sc.nextLine();
        System.out.print("Field              : ");
        field = sc.nextLine();
        System.out.print("Electoral District : ");
        electoralDistrict = sc.nextLine();
        System.out.print("Position           : ");
        position = sc.nextLine();
        System.out.print("Party              : ");
        party = sc.nextLine();

        target.setName(name);
        target.setField(field);
        target.setElectoralDistrict(electoralDistrict);
        target.setPosition(position);
        target.setParty(party);

        System.out.println("Data successfully modified!");

    }

    public void deleteData() {
        String targetID;
        System.out.print("\nPlease select data to delete by ID : ");
        targetID = sc.next();

        AnggotaDPR target = this.searchById(targetID);
        if(target == null) {
            System.err.println("Data was not found!");
        }

        

        System.out.println("Data has been found.");
        System.out.println(target.getId() + ". " + target.getName());
        System.out.println("Confirm to delete? (y/n)");

        char del_confirm = sc.next().charAt(0);

        if(del_confirm == 'y') {
            this.container.remove(target);
        } else {
            System.out.println("Cancelled data deletion\n");
        }
        
        
    }
    
    public void showAll() {
        Table table = new Table();
        int numRows = this.container.size();
        
        String[][] data = new String[numRows + 1][6];
        
        data[0][0] = "ID";
        data[0][1] = "Name";
        data[0][2] = "Field";
        data[0][3] = "Electoral District";
        data[0][4] = "Position";
        data[0][5] = "Party";



        for (int i = 0; i < numRows; i++) {
            AnggotaDPR anggotaDPR = container.get(i);
            data[i + 1][0] = anggotaDPR.getId();
            data[i + 1][1] = anggotaDPR.getName();
            data[i + 1][2] = anggotaDPR.getField();
            data[i + 1][3] = anggotaDPR.getElectoralDistrict();
            data[i + 1][4] = anggotaDPR.getPosition();
            data[i + 1][5] = anggotaDPR.getParty();
        }

        table.createRows(numRows + 1, 6, data);
        
    }

}
